distributed/utils_test.py

```diff
@@ -788,14 +793,21 @@ async def test_foo():
     )
 
     def _(func):
-        def test_func():
+        @functools.wraps(func)
+        def test_func(*outer_args, **kwargs):
             with clean() as loop:
-                if iscoroutinefunction(func):
-                    cor = func
-                else:
-                    cor = gen.coroutine(func)
-                loop.run_sync(cor, timeout=timeout)
 
+                async def coro():
+                    coro = func(*outer_args, **kwargs)
+                    return coro
+
+                loop.run_sync(coro, timeout=timeout)
+
+        # Patch the signature so pytest can inject fixtures
+        orig_sig = inspect.signature(func)
+        test_func.__signature__ = orig_sig.replace(
+            parameters=[p for _, p in orig_sig.parameters.items()]
```

Comment:
This function doesn't take any args, should I do `origin_sig.parameters.values()` instead?

Reply:
Ah, I bet you could even just do `test_func.__signature__ = inspect.signature(func)`. All that signature-munging in `gen_cluster` was because that decorator passes extra things into the test function.

Reply:
That makes a lot of sense , I thought we still had to pass each parameter from the original sig, due to the timeout but I was wrong haha

Changed on the next commit thank you so much for the help!